### PR TITLE
Improve format_level_of_theory 

### PR DESCRIPTION
**Pull request by oscarwumit**
1. consolidated functions related to standardize job level of theory for internal use in ARC.
2. improved function related to format job level of theory for logging purpose.

### CODE REVIEW

**Comment by alongd** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           

Complexity increasing per file
==============================
- arc/commonTest.py  6
         

Complexity decreasing per file
==============================
+ arc/schedulerTest.py  -3
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/ReactionMechanismGenerator/ARC/pullRequest?prid=4762782&bid=15879757)

**Comment by alongd** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           

Complexity increasing per file
==============================
- arc/commonTest.py  6
- arc/job/job.py  2
         

Complexity decreasing per file
==============================
+ arc/schedulerTest.py  -3
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/ReactionMechanismGenerator/ARC/pullRequest?prid=4762782&bid=15879757)

**Comment by alongd** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- arc/job/job.py  2
- arc/commonTest.py  6
         

Complexity decreasing per file
==============================
+ arc/schedulerTest.py  -3
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/ReactionMechanismGenerator/ARC/pullRequest?prid=4762782&bid=15879757)

File: arc/main.py [hunk 1]
```diff
@@ -906,17 +906,17 @@ def determine_model_chemistry_for_job_types(self):
         else:
             default_flag = ''
         self.conformer_level, level = format_level_of_theory_inputs(self.conformer_level)
-        logger.info(f'Using{default_flag} level {level} for refined conformer searches (after filtering via force '
-                    f'fields).')
+        logger.info(f'Using{default_flag} level {format_level_of_theory_for_logging(level)} for refined conformer '
```
Comment (alongd) [inline arc/main.py]: `level` is the output of `format_level_of_theory_inputs`. Do we need to call `format_level_of_theory_for_logging` for it?

File: arc/job/job.py [hunk 2]
```diff
@@ -586,7 +586,7 @@ def write_input_file(self):
         job_options_keywords, job_options_blocks, shortcut_keywords = '', '', ''
 
         # Ignore user specified additional job job_options_keywords when troubleshoot
-        if self.trsh:
+        if self.trsh and (self.job_shortcut_keywords or self.job_shortcut_keywords):
```
Comment (alongd) [inline arc/job/job.py]: maybe
```python
(self.job_additional_options.get(self.software, "") or self.job_shortcut_keywords.get(self.software, ""))
```
instead?

File: arc/common.py [hunk 3]
```diff
@@ -671,6 +671,87 @@ def determine_model_chemistry_type(method):
...
+                                              'auxiliary_basis': 'aug-cc-pVTZ/C cc-pVTZ-F12-CABS'}
+
+    Returns:
+        formatted_model_chemistry_dict (dict): The formatted model chemistry dictionary.
+                                               Default keys: 'method', 'basis', 'auxiliary_basis', 'dispersion'
+        formatted_model_chemistry_str (str): The formatted model chemistry string.
+                                             Format: method|basis|auxiliary_basis|dispersion
+
+    Raises:
+        InputError: If ``model_chem_input`` contains illegal specifications.
```
Comment (alongd) [inline arc/common.py]: replace `model_chem_input` with `level_of_theory`

File: arc/common.py [hunk 4]
```diff
@@ -671,6 +671,87 @@ def determine_model_chemistry_type(method):
     return model_chemistry_class
 
 
+def format_level_of_theory_inputs(level_of_theory):
```
Comment (alongd) [inline arc/common.py]: great!

File: arc/main.py (hunk 1 above)
Comment (oscarwumit) [reply arc/main.py]: yes, in logging we allow things like `dlpno-ccsd(T)/def2-svp/def2-svp/c`, `b3lyp/6-31g gd3bj` which is not allowed by `format_level_of_theory_inputs`

File: arc/job/job.py (hunk 2 above)
Comment (oscarwumit) [reply arc/job/job.py]: ok

File: arc/common.py (hunk 3 above)
Comment (oscarwumit) [reply arc/common.py]: done

**Comment by oscarwumit** [comment]
I have made the changes. @alongd

**Comment by alongd** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- arc/job/job.py  2
- arc/commonTest.py  6
         

Complexity decreasing per file
==============================
+ arc/schedulerTest.py  -3
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/ReactionMechanismGenerator/ARC/pullRequest?prid=4762782&bid=15879757)

**Comment by alongd** [comment]
One more request, please:
Could you add a few examples for the transformations each of the `format_level_of_theory_inputs` and `format_level_of_theory_for_logging` functions make to their docstring? I guess I'm still a bit confused. Thanks!

**Comment by oscarwumit** [comment]
Sure I will add some. Btw, in the tests there are many examples as well.

**Comment by alongd** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- arc/job/job.py  2
- arc/commonTest.py  6
         

Complexity decreasing per file
==============================
+ arc/schedulerTest.py  -3
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/ReactionMechanismGenerator/ARC/pullRequest?prid=4762782&bid=15879757)

**Pull request merged**